engine/shaderc/src/shaderc_private.h

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020-2024 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#ifndef SHADERC_PRIVATE_H
+#define SHADERC_PRIVATE_H
+
+#include "shaderc.h"
+
+#include <spirv/spirv_cross_c.h>
+
+namespace dmShaderc
+{
+    struct ShaderContext
+    {
+        uint8_t*       m_ShaderCode;
+        uint32_t       m_ShaderCodeSize;
+
+        spvc_context   m_SPVCContext;
+        spvc_parsed_ir m_ParsedIR;
+
+        spvc_compiler  m_CompilerNone;
+        spvc_resources m_Resources;
+
+        ShaderReflection m_Reflection;
+    };
+
+    struct ShaderCompiler
+    {
+        ShaderLanguage m_Language;
+    };
+
+    struct ShaderCompilerSPVC
+    {
+        ShaderCompiler m_BaseCompiler;
+        spvc_compiler  m_SPVCCompiler;
+    };
+
+    struct ShaderCompilerSPIRV
+    {
+        struct RemapResourceEntry
+        {
+            uint64_t m_NameHash;
+
+            union Value
+            {
+                uint8_t m_Location;
+                uint8_t m_Binding;
+                uint8_t m_Set;
+                uint8_t m_Value;
+            };
+
+            enum Type
+            {
+                TYPE_LOCATION,
+                TYPE_BINDING,
+                TYPE_SET,
+            };
```

Comment:
Are we missing a TYPE_VALUE?
Or is it confusing that they're not 1:1?

Reply:
Right, yeah I can probably remove the m_Value. I just wanted to set "any of them" using a single value but it doesn't matter.